### PR TITLE
misc: Update .mailmap file

### PR DESCRIPTION
**Pull request by ivanaamit**
The .mailmap file is designed to maintain a record of unique contributors, aiming for a single identifier for each person. What is included in this file does not impact or alter commits; rather, it just merges the counts for all commits by one person under a single name.

### CODE REVIEW

File: .mailmap
Comment (powerjg) [inline .mailmap]: Probably want to remove the comma and swap the name order. I could be wrong, though.

File: .mailmap
Comment (powerjg) [inline .mailmap]: This is Zhantong Qiu (Who may also have a UCD email somewhere...)

File: .mailmap
Comment (mattsinc) [inline .mailmap:267]: I think 276 and 277 are the same except for capitalization?

Comment (ivanaamit) [reply .mailmap:267]: Fixed it. Thanks.

File: .mailmap
Comment (mattsinc) [inline .mailmap:58]: Probably we want line 58 to be fixed as Nilay's email address is embedded in Brad's info?

Also I think 57 and 58 are identical except for this issue.

File: .mailmap
Comment (mattsinc) [inline .mailmap:96]: There are some minor differences in last name (hyphen vs no hyphen) but otherwise 96 and 97 appear to be the same.

File: .mailmap
Comment (mattsinc) [inline .mailmap:118]: Nilay's info embedded in 118 again.

After fixing, 117 and 118 seem the same.

File: .mailmap
Comment (mattsinc) [inline .mailmap:233]: 233 and 234 are identical

File: .mailmap
Comment (mattsinc) [reply .mailmap:267]: I am not sure I fully understand the need to have two very similar versions of my info here, but if this satisfies the Davis folks, fine with me.

File: .mailmap
Comment (mattsinc) [inline .mailmap:311]: Nilay's info has Tim Jones' embedded into it.

File: .mailmap
Comment (mattsinc) [inline .mailmap:337]: 337 and 338 identical.

File: .mailmap
Comment (mattsinc) [inline .mailmap:348]: 348 and 349 identical.

File: .mailmap
Comment (ivanaamit) [reply .mailmap:233]: They are not identical. One is ",edu" and the other one is ".edu".

File: .mailmap
Comment (powerjg) [reply .mailmap:58]: I think there may be some confusion about the purpose/syntax of this file. Here's link to the documentation: https://git-scm.com/docs/gitmailmap

Basically, each line is `[correct info] [incorrect info that should be the info on the left]`.

So, in this case, Brad made a change, but Nilay committed on behalf of brad (because this was how it worked with mercurial before we moved to git). So, the commit was authored by Brad, not Nilay. Nilay was just the one that happened to push upstream (again, mercurial was different from the way git works and the conversion from hg to git wasn't perfect).

So, this is correct. Anytime there is a commit that says "Brad Beckmann ext:(..." we should interpret this as "Brad Beckmann <brad.beckmann@amd.com>".

The main point of this file is so we can track community stats. So, we want to make sure that we have unique (and preferably correct) names on the left hand side of each line. This will allow us to answer questions like "how many people have contributed to the gem5 codebase" or "what is the number of committers with more that 10 commits in the past 2 years."

Finally, I'll point out this *doesn't change the commit info at all*. So, all of the details of which institution is still available. I believe that `git shortlog` and a couple of other commands like that are the only times the mailmap file is used.

File: .mailmap
Comment (powerjg) [reply .mailmap:96]: Right, that's the whole point of this file. This is the same person. So we want `David Guillen <david.guillen@arm.com>`, `David Guillen Fandos <david.guillen@arm.com>`, and `David Guillen-Fandos <david.guillen@arm.com>` to show up as David Guillen-Fandos <david.guillen@arm.com>`. 

We *do not want* these three "different" committers to show up as three different individuals when we run `git shortlog`. We want this to show up as a single entry. This file explains how to relate different committer details to the same human.

File: .mailmap
Comment (mattsinc) [reply .mailmap:233]: The ",edu" one is a typo then, as ",edu" is not a valid email address.

File: .mailmap
Comment (mattsinc) [reply .mailmap:58]: ok, as long as the ones I pointed out that seem duplicated are intentional, then that's fine

File: .mailmap
Comment (powerjg) [reply .mailmap:233]: Right... again, that's the point of this file is to fix these "typos" in the commits. We can't change the commit information, but we can give git a way to know that the email `lena@cs.wisc.edu` and `lena@cs.wisc,edu` are actually the same person, which is also the same as `leolson@google.com`

Comment (mattsinc) [reply .mailmap:233]: I think our messages are just flying past one another, but just acknowledging you answered this with your previous post.

Comment (powerjg) [reply .mailmap:233]: No worries. Thanks for the careful review!

File: .mailmap
Comment (BobbyRBruce) [inline .mailmap:26]: I _think_ there were reasons we kept these capitalizations. Some commits had them and we needed to include these entries to match them (there was some case-sensitivity in the way the author search works in git, maybe (??)). 

Have we proven they are unnecessary?

**Review by BobbyRBruce — COMMENTED** [review]
I just want some clarification about the `ARM.come` deletions. If they are fine, then it's a thumbs-up from me.

File: .mailmap
Comment (ivanaamit) [reply .mailmap:26]: It looks like the mailmap file works case-insensitively. You can check it out [here](https://git-scm.com/docs/gitmailmap).

Comment (andysan) [reply .mailmap:26]: It's worth checking if this actually works the way it's specified in the documentation. I tried removing the different capitalisations locally at one point and it seemed like the `git-shortlog` didn't merge entries correctly at the time.

**Comment by andysan** [comment]
The mailmap file was briefly discussed in PMC last year. However, it seems like the conclusions weren't implemented at the time.

The short summary is that we currently map the tuple name and email to an arbitrary constant mail/email tuple. In some cases, even an outdated email address. This can sometimes be problematic. For example, I have contributed to gem5 in a personal capacity where I personally own the copyright of the commits that get mapped to my Arm email making it seem like these should be copyright Arm. Similarly, there are others who have left Arm and now work on other architectures, but their contributions still get attributed to Arm. What's worse is that their new commits get mapped to an outdated email address they no longer have access to.

It was suggested that we should apply a variant of these rules at the time:
I think we need to come up with some policy for how we manage the mailmap file since this is potentially quite sensitive. In my opinion, we should have rules along these lines:

- Anyone can submit "obvious" fixes where obvious is minor changes such as consistent use of uppercase/lowercase and obvious misspellings.
- Obvious email remapping, such as spelling and short/long form but obviously belonging to the same institution, should be acceptable.
- Changes to a domain name should only be done if requested by the owner of the email address. For example, my old contributions using my personal email address shouldn't be attributed to Arm.
- Name changes (e.g., after getting married) should normally only be done if requested by the affected person. Adding/removing a middle name, spelling, or replacing nicknames would be exceptions to this rule.

We should probably document this somewhere better at some point. Maybe as a discussion or issue?

File: .mailmap
Comment (ivanaamit) [reply .mailmap:26]: We did check before pushing the commit that the file works case-insensitively. Maybe it didn't work that way before.

**Comment by powerjg** [comment]
@andysan, I'd like to kindly remind you that the mailmap file is not a legal document, nor does it alter the commit history when the institution for commits is changed. This point is crucial and, as you may notice from previous discussions in this thread, has been thoroughly clarified: **The file is solely used for statistical purposes**. For instance, it helps in reporting to funders about the growth of our community. The **exclusive purpose** of this file is to track statistics, ensuring that committer information, which remains unaltered, is accurately associated with real individuals.

**Comment by andysan** [comment]
> @andysan, I'd like to kindly remind you that the mailmap file is not a legal document, nor does it alter the commit history when the institution for commits is changed. This point is crucial and, as you may notice from previous discussions in this thread, has been thoroughly clarified: **The file is solely used for statistical purposes**. For instance, it helps in reporting to funders about the growth of our community. The **exclusive purpose** of this file is to track statistics, ensuring that committer information, which remains unaltered, is accurately associated with real individuals.

I understand this. The problem is that it affects how the normal `git-log` command displays commits by default. Unless you actively disable mailmap, it looks like the commit history has changed. It's made worse by the fact that there are known cases where the commit log doesn't show the most up-to-date email address even though the original commit does. This is clearly wrong.

In most cases, there is no need to remap email addresses across domains. The git-shortlog command is able to merge commits from the same author using multiple email addresses (I believe this is the default behaviour) and it supports showing them as separate committers. In fact, I would argue that mapping commits to across institutions is detrimental for statistics since this incorrectly inflates/deflates an institution's contributions to gem5. For example, multiple RISC-V commits are incorrectly being attributed to Arm because of this. This is also clearly wrong.

**Comment by powerjg** [comment]
> > @andysan, I'd like to kindly remind you that the mailmap file is not a legal document, nor does it alter the commit history when the institution for commits is changed. This point is crucial and, as you may notice from previous discussions in this thread, has been thoroughly clarified: **The file is solely used for statistical purposes**. For instance, it helps in reporting to funders about the growth of our community. The **exclusive purpose** of this file is to track statistics, ensuring that committer information, which remains unaltered, is accurately associated with real individuals.
> 
> I understand this. The problem is that it affects how the normal `git-log` command displays commits by default. Unless you actively disable mailmap, it looks like the commit history has changed. It's made worse by the fact that there are known cases where the commit log doesn't show the most up-to-date email address even though the original commit does. This is clearly wrong.
> 
> In most cases, there is no need to remap email addresses across domains. The git-shortlog command is able to merge commits from the same author using multiple email addresses (I believe this is the default behaviour) and it supports showing them as separate committers. In fact, I would argue that mapping commits to across institutions is detrimental for statistics since this incorrectly inflates/deflates an institution's contributions to gem5. For example, multiple RISC-V commits are incorrectly being attributed to Arm because of this. This is also clearly wrong.

I'm pretty sure it's a local option to use the mailmap for git-log. See https://stackoverflow.com/questions/13777171/configuring-git-log-to-use-mailmap-by-default and https://git-scm.com/docs/git-log#Documentation/git-log.txt---no-mailmap

**Review by BobbyRBruce — APPROVED** [review]

**Comment by powerjg** [comment]
I went ahead and merged as-is, because I don't think this changes anything with respect to legal status compared to before it was merged. 

If someone else wants to make updates to this file (or delete it altogether), feel free to create a PR so we can discuss.

**Comment by andysan** [comment]
> > > @andysan, I'd like to kindly remind you that the mailmap file is not a legal document, nor does it alter the commit history when the institution for commits is changed. This point is crucial and, as you may notice from previous discussions in this thread, has been thoroughly clarified: **The file is solely used for statistical purposes**. For instance, it helps in reporting to funders about the growth of our community. The **exclusive purpose** of this file is to track statistics, ensuring that committer information, which remains unaltered, is accurately associated with real individuals.
> > 
> > 
> > I understand this. The problem is that it affects how the normal `git-log` command displays commits by default. Unless you actively disable mailmap, it looks like the commit history has changed. It's made worse by the fact that there are known cases where the commit log doesn't show the most up-to-date email address even though the original commit does. This is clearly wrong.
> > In most cases, there is no need to remap email addresses across domains. The git-shortlog command is able to merge commits from the same author using multiple email addresses (I believe this is the default behaviour) and it supports showing them as separate committers. In fact, I would argue that mapping commits to across institutions is detrimental for statistics since this incorrectly inflates/deflates an institution's contributions to gem5. For example, multiple RISC-V commits are incorrectly being attributed to Arm because of this. This is also clearly wrong.
> 
> I'm pretty sure it's a local option to use the mailmap for git-log. See https://stackoverflow.com/questions/13777171/configuring-git-log-to-use-mailmap-by-default and https://git-scm.com/docs/git-log#Documentation/git-log.txt---no-mailmap

Yes, there is and it seems to use mailmap by default on Ubuntu 22.04.

I have looked a bit at how this is done in other projects. Linux makes extensive use of mailmap and Zephyr seems to use it as well. In both of these projects, it seems like the assumption is that emails follow the author (the sign-off line is what is used to identify copyright) most of the time. I.e., mailmap maps to the most recent valid email address. I think that's what we are trying to achieve here. However, there are multiple cases where we map correct email addresses to incorrect email addresses that are no longer in use which seems wrong.

It also seems like mailmap is almost exclusively updated by the people affected by the mapping. That's probably not feasible in gem5 though.

**Pull request merged**
